src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,103 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int threshold;
+    private Node<K, V>[] table;
+    private int size;
+    private int capacity;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+        capacity = DEFAULT_CAPACITY;
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(key, value, null);
+        Node<K, V> currentNode = table[getIndex(key)];
+        if (currentNode == null) {
+            size++;
+            resizeIfNeeded();
+            table[getIndex(key)] = newNode;
+            return;
+        }
+        if (Objects.equals(key, currentNode.key)) {
+            newNode.next = currentNode.next;
+            table[getIndex(key)] = newNode;
+            return;
+        }
+        while (currentNode.next != null) {
+            if (Objects.equals(key, currentNode.next.key)) {
+                newNode.next = currentNode.next.next;
+                currentNode.next = newNode;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
+        size++;
+        if (resizeIfNeeded()) {
+            put(newNode.key, newNode.value);
+        }
+        currentNode.next = newNode;
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> currentNode = table[getIndex(key)];
+        while (currentNode != null) {
+            if (Objects.equals(key, currentNode.key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private boolean resizeIfNeeded() {
```

Comment:
This method should be private because it's a utility method and used only inside the same class. Remember to follow the encapsulation principle.